common/services/population.js

```diff
@@ -51,6 +51,25 @@ class PopulationService extends BaseService {
       .then(response => response.data)
   }
 
+  populate(data) {
+    const { location, date } = data
+
+    if (!location) {
+      return Promise.reject(new Error(noLocationIdMessage))
+    }
+
+    if (!date) {
+      return Promise.reject(new Error(noDateMessage))
+    }
+
+    return this.apiClient
```

Comment:
The endpoint to populate new data isn't JSON api compliant - see #1269 for further details.

In order to fetch the data, I've needed to drop down into the base `request` method.

Reply:
What was the reason for just a bespoke endpoint? I had a look at [1269](https://github.com/ministryofjustice/hmpps-book-secure-move-api/pull/1269) on the API but couldn't see anything around making this a normal REST endpoint.

Also, [the documentation](https://hmpps-book-secure-move-api-staging.apps.live-1.cloud-platform.service.justice.gov.uk/api-docs/index.html#operations-Populations-get_populations_new) suggests that it does contain a JSON:API payload?

Reply:
The problem I encountered was because I wasn't specifying an ID, `JSON` api **helpfully** pluralised the word for me so made a request to `/population/news` not `/population/new`

Reply:
Ah, I see, it was the endpoint and not the payload. Devour does this based on the global setting for [pluralisation](https://github.com/twg/devour#initializer) as the expectation is that endpoints are consistently named.

You do however, have two other options in this instance which might be neater:

1. In the model options set `pluralize` to `false`
2. In the model options use a custom `collectionPath` with the singular form (`/population/new`). We do a similar thing for the [reference data models](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/c35022a6b2924599878028a1e8740039c6cfbf69/common/lib/api-client/models/gender.js#L11)

Reply:
Ok, I'll give that a try!

I do need a combination of pluralised (`/populations/` / `/populations/{id}`) and non-plurarlised (`/populations/new`) endpoints so hopefully some thing better than what I've got will work :)

Reply:
Ah, I see what you mean. In fact, maybe what you have is actually easiest then.